sdk/python/kfp/containers/_component_builder.py

```diff
@@ -229,16 +284,21 @@ def build_python_component(
     base_image = default_base_image_or_builder
     if isinstance(base_image, Callable):
       base_image = base_image()
+  if not dependency:
+    dependency = []
```

Comment:
This is because mutable default is discouraged in Python and can potentially cause surprising behavior.

See See for example, https://nikos7am.com/posts/mutable-default-arguments/#:~:text=Avoid%20using%20an%20empty%20list%20as%20a%20default%20argument%20to%20a%20function,-June%204%2C%202018&text=This%20is%20not%20the%20behavior,time%20the%20function%20is%20called.

Reply:
That's good to learn. Thanks for the reference.